### PR TITLE
[RL] Expose top-p-only sampling masks

### PR DESCRIPTION
**Pull request by nanjiangwill**
## Relationship to existing work

PR #27408 introduced SGLang's native opt-in sampling-mask primitive and exposed it through `/generate` for requests with a finite `top_k`.

This PR completes the request surface by admitting top-p-only masks on `/generate` and exposing `return_sampling_mask` through `/v1/chat/completions`. General sampler and scheduler hardening is intentionally separate in #34037.

## Motivation

The existing request admission requires a finite `top_k`. This excludes the common RL rollout configuration `top_k=-1` with `top_p<1`, even though top-p truncation defines the replayable sampling support. Chat completions also do not expose the primitive, while Miles uses both `/generate` and `/v1/chat/completions`.

## Endpoint scope

| Endpoint | Before | After |
| --- | --- | --- |
| `/generate` | Sampling masks require finite `top_k` | Top-p-only masks are admitted |
| `/v1/chat/completions` | Sampling-mask option is not exposed | `return_sampling_mask` is exposed when `return_meta_info=true` |
| `/v1/completions` | Sampling-mask option is not exposed | Unchanged; out of scope |

Both supported endpoints use the existing SGLang sampling-mask primitive and return `output_token_sampling_mask` plus `output_token_sampling_logprobs` metadata.

## Modifications

- Admit a sampling-mask request when either `top_k` is finite or `top_p` is less than 1.
- Continue rejecting an explicit full-vocabulary mask request when `top_k` is unlimited and `top_p` is 1.
- Add `return_sampling_mask` to the chat-completions request schema.
- Forward the chat option to the existing `GenerateReqInput.return_sampling_mask` field.
- Require `return_meta_info=true` for chat requests so the requested metadata is included in the response.

## Tests

Coverage includes:

- top-p-only masks through `/generate`;
- top-p-only masks through `/v1/chat/completions`;
- full-vocabulary admission rejection;
- response mask/token alignment;
- chat request defaults, validation, and internal conversion.

Local checks:

- protocol and serving-chat unit suites: 135 passed, including 54 subtests;
- repository-native all-files `pre-commit` pipeline passed;
- `git diff --check`;
- Python bytecode compilation for every changed Python file.

GPU endpoint coverage is registered for CI and was not run locally.

## Non-goals

- No sampling-mask reconstruction or performance changes.
- No new sampling algorithm or change to sampling semantics.
- No change when `return_sampling_mask` is false.
- No `/v1/completions` support.
- No speculative-decoding or Ascend sampling-mask support.

---
### CI States

Latest PR Test (Base): :white_check_mark: [Run #31470462930](https://github.com/sgl-project/sglang/actions/runs/31470462930)
Latest PR Test (Extra): :x: [Run #31470462865](https://github.com/sgl-project/sglang/actions/runs/31470462865)

### CODE REVIEW

**Review by guapisolo — APPROVED** [review]
LGTM.

File: python/sglang/srt/managers/scheduler.py [hunk 1]
```diff
@@ -2460,11 +2460,13 @@ def handle_generate_request(
             self._add_request_to_queue(req)
             return
 
-        if req.return_sampling_mask and req.sampling_params.top_k == TOP_K_ALL:
+        uses_top_k_or_top_p_truncation = (
+            req.sampling_params.top_k != TOP_K_ALL or req.sampling_params.top_p < 1.0
```
Comment (JustinTong0323) [inline python/sglang/srt/managers/scheduler.py:2521]: Allowing `TOP_K_ALL` here makes any top-p-only mask request set `sampling_mask_max_top_k` to `TOP_K_ALL`; `Sampler._compute_sampling_mask_from_probs` then full-sorts every row in the decode batch and materializes all kept IDs on CPU before filtering `return_sampling_masks`, so one opt-in request can impose full-vocabulary work on unrelated co-batched requests. Restrict reconstruction and host transfer to requesting rows, and add a hard cap on the realized support size. `top_p < 1` is not a size bound: flat logits retain roughly `top_p * vocab_size`, and near-1 values can round to `1.0` in the float32 `top_ps` tensor, so this condition can still return a full or nearly full vocabulary.

**Review by JustinTong0323 — COMMENTED** [review]
The top-p-only path needs per-request isolation and a hard support-size limit before it is safe to enable.

File: python/sglang/srt/managers/scheduler.py (hunk 1 above)
Comment (nanjiangwill) [reply python/sglang/srt/managers/scheduler.py:2521]: the performance concern is valid. a top-p-only request can still have a large support and make reconstruction expensive for the rest of the batch. But that is a general issue with the primitive from #27408(large finite-`top_k` requests have the same problem) rather than something specific to exposing top-p here.

I split the row filtering and size bound into #34037 because those changes belong in the primitive and apply to both #27408 and this PR. I think this condition should stay here: it is the missing admission piece for top-p-only requests, and #33593 can merge independently before the general hardening follow-up.

**Review by JustinTong0323 — APPROVED** [review]
LGTM
